### ISSUE TITLE
getSubdomain does not perform as expected for .er top level domains

### ISSUE THREAD

**Issue by nmargaritis** [oncletom/tld.js#78]
Testing the following host "shop.google.er" will return an empty subdomain. For example,
`
tldjs.getSubdomain("shop.google.er"); // ''
tldjs.getDomain("shop.google.er"); // shop.google.er
tldjs.isValid("shop.google.er"); // true
tldjs.tldExists("shop.google.er"); // true
`
Although the expected outcome would be to have a subdomain of _shop_ and a domain of _google.er_.

**Comment by oncletom** [comment]
@nmargaritis from the publicsuffixlist, I see that `er` is declared as `*.er`.

The confusion lies in #25 where the rules are defined for cookie parsing – so technically speaking `shop.google.er` is not a valid subdomain for the `er` tld.

It is very likely to be addressed by adding/renaming methods to make them explicit in term of what they are trying to do.

**Comment by remusao** [comment]
@nmargaritis @oncletom Should we close this issue, as this is expected behavior?

**Comment by nmargaritis** [comment]
@remusao @oncletom For some reason i missed the initial reply. If this is the expected behaviour then to my understanding the function **isValid** should fail instead of returning true.

**Comment by remusao** [comment]
@nmargaritis From what I understood, the rule from public suffix is `*.er`, which means that `google.er` is a valid public suffix, hence `isValid` returns `true` for `shop.google.er`.

**Comment by nmargaritis** [comment]
@remusao Yes that is correct for google.er and .er public suffix but isValid is described as "Is the hostname valid according to the RFC". So it should fail for **shop.google.er** since it is not valid based on the spec.
